### PR TITLE
Improving fetching of node versions [semver:minor]

### PR DESCRIPTION
**Pull request by michaelgmcd**
### Checklist

- [X] All new jobs, commands, executors, parameters have descriptions
- [X] Examples have been added for any significant new features
- [X] README has been updated, if necessary

### Motivation, issues

- Fixes a bug fetching the  "latest" node version that could cause build failures.
- Most users will want the "lts" or "stable" version

### Description

- The current method of getting the latest Node.js is incorrect. The "latest" version currently fetched is the most recently updated, which is incorrect. At the time of this commit, the "latest" version on GitHub is 10.20.1, well behind the latest.
- Adds support for "lts" and "stable" as node-version parameters

This new method is based on tj/n's method of fetching versions:
https://github.com/tj/n/blob/8c55fba3d504a69e79946dfef1ad242c67f27552/bin/n#L960-L1019

tj/n references Node's own dist-indexer as a source:
https://github.com/nodejs/nodejs-dist-indexer/blob/master/dist-indexer.js

### CODE REVIEW

**Comment by michaelgmcd** [comment]
So the tests are failing, but it's a false negative.
Latest on the GitHub releases is 13.14.0: https://github.com/nodejs/node/releases
Latest on the Node.js website is 14.1.0: https://nodejs.org/en/

This PR pulls 14.1.0 from Node.js directly.

**Comment by lokst** [comment]
@michaelgmcd Thank you for your changes! We will look into updating the test to make the build pass.

File: src/commands/install.yml [hunk 1]
```diff
@@ -60,11 +61,18 @@ steps:
...
+
         get_node_version () {
           if [[ <<parameters.node-version>> == "" || <<parameters.node-version>> == "latest" || <<parameters.node-version>> == "current" ]]; then
-            NODE_ORB_INSTALL_VERSION=$(curl -Ls -o /dev/null -w %{url_effective} \
-              "https://github.com/nodejs/node/releases/latest" | sed 's:.*/::' | cut -d 'v' -f 2)
+            NODE_ORB_INSTALL_VERSION=$(get_node_index | awk "NR<=1" | cut -f 1 | grep -E -o '[^v].*')
             echo "Latest version of Node.js is $NODE_ORB_INSTALL_VERSION"
+          elif [[ <<parameters.node-version>> == "lts" || <<parameters.node-version>> == "stable" ]]; then
+            # Codename is last field, first one with a name is newest lts
+            NODE_ORB_INSTALL_VERSION=$(get_node_index | grep -E "\t[a-zA-Z]+\$" | awk "NR<=1" | cut -f 1 | grep -E -o '[^v].*')
```
Comment (lokst) [inline src/commands/install.yml:74]: @michaelgmcd It looks like this passes on macos (https://app.circleci.com/pipelines/github/CircleCI-Public/node-orb/461/workflows/9a88f65c-95fb-4b4e-a308-5e8a0ea62632/jobs/3201) but not on linux (https://app.circleci.com/pipelines/github/CircleCI-Public/node-orb/461/workflows/9a88f65c-95fb-4b4e-a308-5e8a0ea62632/jobs/3199), would you mind taking a look?

Comment (michaelgmcd) [reply src/commands/install.yml:74]: Tried investigating, but the only log is "Exited with status code 1"

Is there a way we could get more error information?

Comment (lokst) [reply src/commands/install.yml]: I think the issue was with the `\t` character used in the grep expression. I managed to resolve the build error by modifying the expression.

**Pull request merged**